client/reader/id-helpers.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * External Dependencies
+ */
+
+/**
+ * Internal Dependencies
+ */
+
+export function toValidId( val ) {
+	if ( val === true ) {
+		return undefined;
+	}
```

Comment:
The RegExp was only there because they are generally really fast but it'd be the same to just check if a dot exists. `Number( 1 / 0 )` returns `Infinity` so it's possible whether or not likely